module/x/gravity/client/cli/query.go

```diff
@@ -351,6 +389,80 @@ func CmdGetAttestations() *cobra.Command {
 	return cmd
 }
 
+// CmdGetERC721Attestations fetches the most recently created Attestations in the store (only the most recent 1000 are available)
+// up to an optional limit
+func CmdGetERC721Attestations() *cobra.Command {
+	short := "Query gravity current and historical erc721 attestations (only the most recent 1000 are stored)"
+	long := short + "\n\n" + "Optionally provide a limit to reduce the number of attestations returned" + "\n" +
+		"Note that when querying with --height less than 1282013 '--use-v1-key' must be provided to locate the attestations"
+
+	cmd := &cobra.Command{
+		Use:   "erc721-attestations [optional limit]",
+		Args:  cobra.MaximumNArgs(1),
+		Short: short,
+		Long:  long,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			var limit uint64
+			// Limit is 0 or whatever the user put in
+			if len(args) == 0 || args[0] == "" {
+				limit = 0
+			} else {
+				limit, err = strconv.ParseUint(args[0], 10, 64)
+				if err != nil {
+					return err
+				}
+			}
+			orderBy, err := cmd.Flags().GetString(FlagOrder)
+			if err != nil {
+				return err
+			}
+			claimType, err := cmd.Flags().GetString(FlagClaimType)
+			if err != nil {
+				return err
+			}
+			nonce, err := cmd.Flags().GetUint64(FlagNonce)
+			if err != nil {
+				return err
+			}
+			height, err := cmd.Flags().GetUint64(FlagEthHeight)
+			if err != nil {
+				return err
+			}
+
+			req := &types.QueryERC721AttestationsRequest{
+				Limit:     limit,
+				OrderBy:   orderBy,
+				ClaimType: claimType,
+				Nonce:     nonce,
+				Height:    height,
+			}
+			res, err := queryClient.GetERC721Attestations(cmd.Context(), req)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	// Global flags
+	flags.AddQueryFlagsToCmd(cmd)
+	// Local flags
+	cmd.Flags().String(FlagOrder, "asc", "order attestations by eth block height: set to 'desc' for reverse ordering")
+	cmd.Flags().String(FlagClaimType, "", "which types of claims to filter, empty for all or one of: CLAIM_TYPE_SEND_TO_COSMOS, CLAIM_TYPE_BATCH_SEND_TO_ETH, CLAIM_TYPE_ERC20_DEPLOYED, CLAIM_TYPE_LOGIC_CALL_EXECUTED, CLAIM_TYPE_VALSET_UPDATED")
```

Comment:
```suggestion
	cmd.Flags().String(FlagClaimType, "", "which types of claims to filter, empty for all or one of: CLAIM_TYPE_SEND_ERC721_TO_COSMOS")
```
Only one claim type for now for erc721